subsys/logging/log_rpc.c

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file
+ *   This file implements Remote Procedure Calls (RPC) for logging subsystem.
+ *
+ */
+
+#include <nrf_rpc_cbor.h>
+#include <nrf_rpc/nrf_rpc_uart.h>
+
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_backend.h>
+#include <zephyr/logging/log_backend_std.h>
+#include <zephyr/logging/log_output.h>
+
+#include <string.h>
+
+#ifdef CONFIG_NRF_RPC_IPC_SERVICE
+NRF_RPC_IPC_TRANSPORT(log_rpc_tr, DEVICE_DT_GET(DT_NODELABEL(ipc0)), "log_rpc_ept");
+#elif defined(CONFIG_NRF_RPC_UART_TRANSPORT)
+#define log_rpc_tr NRF_RPC_UART_TRANSPORT(DT_NODELABEL(uart1))
+#endif
+NRF_RPC_GROUP_DEFINE(log_rpc_group, "log", &log_rpc_tr, NULL, NULL, NULL);
+
+enum log_rpc_evt_forwarder {
+	LOG_RPC_EVT_MSG = 0,
```

Comment:
Enum initializing to zero is a bit useless since it starts from zero according to C language.

Reply:
Right, though for things like message IDs it's useful to have all values assigned explicitly for faster lookups (in the logs you only have the numeric representation available).